packages/plugins/typescript/src/index.ts

```diff
@@ -22,6 +36,32 @@ export const plugin: PluginFunction<TypeScriptPluginConfig> = (
   const astNode = parse(printedSchema);
   const header = `type Maybe<T> = ${visitor.config.maybeValue};`;
   const visitorResult = visit(astNode, { leave: visitor });
+  const introspectionDefinitions = includeIntrospectionDefinitions(schema, documents, config);
 
-  return [header, ...visitorResult.definitions].join('\n');
+  return [header, ...visitorResult.definitions, ...introspectionDefinitions].join('\n');
 };
+
+function includeIntrospectionDefinitions(
+  schema: GraphQLSchema,
+  documents: DocumentFile[],
+  config: TypeScriptPluginConfig
+): string[] {
+  const typeInfo = new TypeInfo(schema);
+  const typesToInclude: GraphQLNamedType[] = [];
+  const documentsVisitor = visitWithTypeInfo(typeInfo, {
+    Field() {
+      const type = getNamedType(typeInfo.getType());
+
+      if (isIntrospectionType(type) && isEnumType(type) && !typesToInclude.includes(type)) {
+        typesToInclude.push(type);
+      }
+    }
+  });
+
+  documents.forEach(doc => visit(doc.content, documentsVisitor));
+
+  const visitor = new TsIntrospectionVisitor(typesToInclude, new TsVisitor(config));
+  const result: DocumentNode = visit(parse(printIntrospectionSchema(schema)), { leave: visitor });
+
+  return result.definitions as any;
```

Comment:
Isn't it an array? why `any?

Reply:
It should be `string[]` but since we convert bunch of things then `DocumentNode` is no longer valid because `result.definitions` is `DefinitionNode[]` which is no longer true.

Reply:
so either `result: any` or `result.definitions any`

Reply:
```suggestion
  return result.definitions as any[];
```